BUG: signal: crash in `signaltools` on free-threaded Python, macOS arm64

In the cron wheel builds on `main`, there was a failure today in the arm64 macOS 14.0 wheel build. The test suite run exited in `signal/tests/test_signaltools.py` with:
```
13615 Trace/BPT trap: 5
```

Full [CI log](https://github.com/scipy/scipy/actions/runs/9854340015/job/27206883516) and raw output (doesn't show anything more informative it looks like):

<details>

```

...


2024-07-09T10:14:17.6854550Z + /bin/sh -c 'bash /Users/runner/work/scipy/scipy/tools/wheels/cibw_test_command.sh /Users/runner/work/scipy/scipy'
2024-07-09T10:14:17.6901730Z ++ python '-cimport sysconfig; print(bool(sysconfig.get_config_var('\''Py_GIL_DISABLED'\'')))'
2024-07-09T10:14:17.7121210Z + FREE_THREADED_BUILD=True
2024-07-09T10:14:17.7121520Z + [[ True == \T\r\u\e ]]
2024-07-09T10:14:17.7122050Z + export PYTHON_GIL=0
2024-07-09T10:14:17.7122330Z + PYTHON_GIL=0
2024-07-09T10:14:17.7122800Z + python -c 'import sys; import scipy; sys.exit(not scipy.test())'
2024-07-09T10:14:25.2555010Z ============================= test session starts ==============================
2024-07-09T10:14:25.2556010Z platform darwin -- Python 3.13.0b2, pytest-8.2.2, pluggy-1.5.0
2024-07-09T10:14:25.2556740Z rootdir: /private/var/folders/h9/l1shxhdd69nct08ylzq0n6q00000gn/T/cibw-run-9m42f5g1/cp313t-macosx_arm64/test_cwd
2024-07-09T10:14:25.2557380Z plugins: hypothesis-6.105.1, cov-5.0.0, xdist-3.6.1
2024-07-09T10:14:25.2557760Z collected 64837 items / 11739 deselected / 53098 selected
2024-07-09T10:14:25.2558000Z 
2024-07-09T10:14:25.2607920Z _lib/tests/test__gcutils.py ......                                       [  0%]
2024-07-09T10:14:25.2682730Z _lib/tests/test__pep440.py .........                                     [  0%]
2024-07-09T10:14:25.2700990Z _lib/tests/test__testutils.py ..                                         [  0%]
2024-07-09T10:14:25.7786640Z _lib/tests/test__threadsafety.py ..                                      [  0%]
2024-07-09T10:14:26.5096540Z _lib/tests/test__util.py ........................                        [  0%]
2024-07-09T10:14:26.5173400Z _lib/tests/test_array_api.py ssssssssssssss                              [  0%]
2024-07-09T10:14:26.5402140Z _lib/tests/test_bunch.py ............................                    [  0%]
2024-07-09T10:14:26.9057650Z _lib/tests/test_ccallback.py ss..                                        [  0%]
2024-07-09T10:14:26.9207020Z _lib/tests/test_deprecation.py .                                         [  0%]
2024-07-09T10:14:27.0253750Z _lib/tests/test_public_api.py .......................................... [  0%]
2024-07-09T10:14:27.1161830Z ...........................................................              [  0%]
2024-07-09T10:14:27.1171160Z _lib/tests/test_scipy_version.py .                                       [  0%]
2024-07-09T10:14:27.1208080Z _lib/tests/test_tmpdirs.py ...                                           [  0%]
2024-07-09T10:14:27.2026800Z cluster/tests/test_disjoint_set.py .....................                 [  0%]
2024-07-09T10:14:27.2528020Z cluster/tests/test_hierarchy.py ........................................ [  0%]
2024-07-09T10:14:27.3202600Z .........................................sss...........                  [  0%]
2024-07-09T10:14:27.7931330Z cluster/tests/test_vq.py ...............................                 [  0%]
2024-07-09T10:14:27.7990700Z constants/tests/test_codata.py .......                                   [  0%]
2024-07-09T10:14:27.8054610Z constants/tests/test_constants.py .......                                [  0%]
2024-07-09T10:14:28.4682470Z datasets/tests/test_data.py .....                                        [  0%]
2024-07-09T10:14:28.5271930Z fft/_pocketfft/tests/test_basic.py ..................................... [  0%]
2024-07-09T10:14:28.6184940Z ........................................................................ [  0%]
2024-07-09T10:14:28.6871320Z ........................................................................ [  1%]
2024-07-09T10:14:28.7571510Z ........................................................................ [  1%]
2024-07-09T10:14:28.8555130Z ........................................................................ [  1%]
2024-07-09T10:14:29.1293840Z ........................................................................ [  1%]
2024-07-09T10:14:29.5032370Z ..............................................................           [  1%]
2024-07-09T10:14:29.5362690Z fft/_pocketfft/tests/test_real_transforms.py ........................... [  1%]
2024-07-09T10:14:29.6031280Z ........................................................................ [  1%]
2024-07-09T10:14:29.6720620Z ........................................................................ [  1%]
2024-07-09T10:14:29.7397130Z ........................................................................ [  2%]
2024-07-09T10:14:29.8047270Z ........................................................................ [  2%]
2024-07-09T10:14:30.1155580Z ........................................................................ [  2%]
2024-07-09T10:14:30.3840320Z ........................................................................ [  2%]
2024-07-09T10:14:30.5076490Z ........................................................................ [  2%]
2024-07-09T10:14:30.5846320Z ........................................................................ [  2%]
2024-07-09T10:14:30.6565160Z ........................................................................ [  2%]
2024-07-09T10:14:30.7249790Z ........................................................................ [  2%]
2024-07-09T10:14:30.8180390Z ........................................................................ [  3%]
2024-07-09T10:14:30.8948810Z ........................................................................ [  3%]
2024-07-09T10:14:31.2646790Z ........................................................................ [  3%]
2024-07-09T10:14:31.5765810Z ........................................................................ [  3%]
2024-07-09T10:14:31.6457940Z ........................................................................ [  3%]
2024-07-09T10:14:31.7139080Z ........................................................................ [  3%]
2024-07-09T10:14:31.7755250Z ........................................................................ [  3%]
2024-07-09T10:14:31.8335770Z ........................................................................ [  4%]
2024-07-09T10:14:31.8937290Z ........................................................................ [  4%]
2024-07-09T10:14:31.9552580Z ........................................................................ [  4%]
2024-07-09T10:14:32.0153930Z ........................................................................ [  4%]
2024-07-09T10:14:32.0742110Z ........................................................................ [  4%]
2024-07-09T10:14:32.1342620Z ........................................................................ [  4%]
2024-07-09T10:14:32.1948590Z ........................................................................ [  4%]
2024-07-09T10:14:32.2564340Z ........................................................................ [  4%]
2024-07-09T10:14:32.3222580Z ........................................................................ [  5%]
2024-07-09T10:14:32.3705760Z .................................................                        [  5%]
2024-07-09T10:14:32.4125240Z fft/tests/test_backend.py .............................................. [  5%]
2024-07-09T10:14:32.4125710Z                                                                          [  5%]
2024-07-09T10:14:32.5495700Z fft/tests/test_basic.py ................................................ [  5%]
2024-07-09T10:14:32.7200660Z ........................................................................ [  5%]
2024-07-09T10:14:33.7593160Z ...................                                                      [  5%]
2024-07-09T10:14:33.8039110Z fft/tests/test_fftlog.py ........................................        [  5%]
2024-07-09T10:14:33.8371100Z fft/tests/test_helper.py ...............................                 [  5%]
2024-07-09T10:14:33.9503940Z fft/tests/test_multithreading.py ....................................... [  5%]
2024-07-09T10:14:33.9837950Z ................                                                         [  5%]
2024-07-09T10:14:34.0240480Z fft/tests/test_real_transforms.py ...................................... [  5%]
2024-07-09T10:14:34.0973000Z ........................................................................ [  6%]
2024-07-09T10:14:34.1705940Z ........................................................................ [  6%]
2024-07-09T10:14:34.2434150Z ........................................................................ [  6%]
2024-07-09T10:14:34.3157850Z ........................................................................ [  6%]
2024-07-09T10:14:34.3873810Z ........................................................................ [  6%]
2024-07-09T10:14:34.4592810Z ........................................................................ [  6%]
2024-07-09T10:14:34.5306500Z ........................................................................ [  6%]
2024-07-09T10:14:34.6022770Z ........................................................................ [  6%]
2024-07-09T10:14:34.6741650Z ........................................................................ [  7%]
2024-07-09T10:14:34.7459530Z ........................................................................ [  7%]
2024-07-09T10:14:34.8142910Z ........................................................................ [  7%]
2024-07-09T10:14:34.8838600Z ........................................................................ [  7%]
2024-07-09T10:14:34.9501190Z ........................................................................ [  7%]
2024-07-09T10:14:35.0187220Z ........................................................................ [  7%]
2024-07-09T10:14:35.0876930Z ........................................................................ [  7%]
2024-07-09T10:14:35.1595300Z ........................................................................ [  8%]
2024-07-09T10:14:35.2356300Z ........................................................................ [  8%]
2024-07-09T10:14:35.3065520Z ........................................................................ [  8%]
2024-07-09T10:14:35.3771920Z ........................................................................ [  8%]
2024-07-09T10:14:35.4550810Z ........................................................................ [  8%]
2024-07-09T10:14:35.5306390Z ........................................................................ [  8%]
2024-07-09T10:14:35.6093410Z ........................................................................ [  8%]
2024-07-09T10:14:35.6897450Z ........................................................................ [  8%]
2024-07-09T10:14:35.7699340Z ........................................................................ [  9%]
2024-07-09T10:14:35.8533310Z ........................................................................ [  9%]
2024-07-09T10:14:35.9350320Z ........................................................................ [  9%]
2024-07-09T10:14:36.0177350Z ........................................................................ [  9%]
2024-07-09T10:14:36.0906920Z ........................................................................ [  9%]
2024-07-09T10:14:36.1640020Z ........................................................................ [  9%]
2024-07-09T10:14:36.2412880Z ........................................................................ [  9%]
2024-07-09T10:14:36.3201230Z ........................................................................ [ 10%]
2024-07-09T10:14:36.3957790Z ........................................................................ [ 10%]
2024-07-09T10:14:36.4712290Z ........................................................................ [ 10%]
2024-07-09T10:14:36.5504580Z ........................................................................ [ 10%]
2024-07-09T10:14:36.6275450Z ........................................................................ [ 10%]
2024-07-09T10:14:36.6996650Z ........................................................................ [ 10%]
2024-07-09T10:14:36.7691050Z ........................................................................ [ 10%]
2024-07-09T10:14:36.8371680Z ........................................................................ [ 11%]
2024-07-09T10:14:36.9076270Z ........................................................................ [ 11%]
2024-07-09T10:14:36.9742240Z ........................................................................ [ 11%]
2024-07-09T10:14:37.0389130Z ........................................................................ [ 11%]
2024-07-09T10:14:37.0908880Z ..........................................................               [ 11%]
2024-07-09T10:14:37.2353020Z fftpack/tests/test_basic.py ......x..................................... [ 11%]
2024-07-09T10:14:37.3292180Z ........................................................................ [ 11%]
2024-07-09T10:14:37.4028710Z ........................................................................ [ 11%]
2024-07-09T10:14:37.6076430Z ........................................................................ [ 12%]
2024-07-09T10:14:37.8148540Z ....................................                                     [ 12%]
2024-07-09T10:14:37.8188790Z fftpack/tests/test_helper.py ....                                        [ 12%]
2024-07-09T10:14:37.8864940Z fftpack/tests/test_pseudo_diffs.py .............................         [ 12%]
2024-07-09T10:14:38.1308650Z fftpack/tests/test_real_transforms.py .................................. [ 12%]
2024-07-09T10:14:38.8570470Z ........................................................................ [ 12%]
2024-07-09T10:14:38.9197940Z ........................................................................ [ 12%]
2024-07-09T10:14:38.9647940Z ..............................................                           [ 12%]
2024-07-09T10:14:39.5883850Z integrate/_ivp/tests/test_ivp.py ....................................... [ 12%]
2024-07-09T10:14:39.7089890Z ...............                                                          [ 12%]
2024-07-09T10:14:39.7182320Z integrate/_ivp/tests/test_rk.py ..........                               [ 12%]
2024-07-09T10:14:41.5625470Z integrate/tests/test__quad_vec.py ..........................             [ 12%]
2024-07-09T10:14:41.6248260Z integrate/tests/test_banded_ode_solvers.py .                             [ 12%]
2024-07-09T10:14:41.7764800Z integrate/tests/test_bvp.py ................                             [ 12%]
2024-07-09T10:14:41.8280720Z integrate/tests/test_integrate.py ...................................... [ 12%]
2024-07-09T10:14:41.8478350Z .........                                                                [ 12%]
2024-07-09T10:14:41.8761530Z integrate/tests/test_odeint_jac.py ..                                    [ 12%]
2024-07-09T10:14:42.0584870Z integrate/tests/test_quadpack.py ....................................... [ 12%]
2024-07-09T10:14:43.0279540Z ..sssssssssssssssssssssssssssssssssssssssssss...........                 [ 13%]
2024-07-09T10:14:43.1334810Z integrate/tests/test_quadrature.py ..................................... [ 13%]
2024-07-09T10:14:43.2340040Z ........................................................................ [ 13%]
2024-07-09T10:14:43.3278030Z ........................................................................ [ 13%]
2024-07-09T10:14:43.4222890Z ........................................................................ [ 13%]
2024-07-09T10:14:43.4286090Z .......                                                                  [ 13%]
2024-07-09T10:14:43.5115680Z integrate/tests/test_tanhsinh.py ....................................... [ 13%]
2024-07-09T10:14:43.6967280Z .........ssss........................................................... [ 13%]
2024-07-09T10:14:44.0721280Z .............................ss..ss......ss............................. [ 13%]
2024-07-09T10:14:44.2780360Z .................................ss..................................... [ 14%]
2024-07-09T10:14:44.4808950Z ........................................                                 [ 14%]
2024-07-09T10:14:44.5441160Z interpolate/tests/test_bsplines.py ..................................... [ 14%]
2024-07-09T10:14:44.6400190Z ........................................................................ [ 14%]
2024-07-09T10:14:45.0000690Z .......................x................................................ [ 14%]
2024-07-09T10:14:45.0590470Z ..............................                                           [ 14%]
2024-07-09T10:14:45.1476160Z interpolate/tests/test_fitpack.py .....................................s [ 14%]
2024-07-09T10:14:45.1579000Z ....s..                                                                  [ 14%]
2024-07-09T10:14:45.3876110Z interpolate/tests/test_fitpack2.py ..................................... [ 14%]
2024-07-09T10:14:45.5359100Z ................................                                         [ 14%]
2024-07-09T10:14:45.5366470Z interpolate/tests/test_gil.py s                                          [ 14%]
2024-07-09T10:14:45.6146460Z interpolate/tests/test_interpnd.py .....................                 [ 14%]
2024-07-09T10:14:45.7301320Z interpolate/tests/test_interpolate.py .................................. [ 14%]
2024-07-09T10:14:46.0830570Z ........................................................................ [ 14%]
2024-07-09T10:14:46.4576170Z ........................                                                 [ 14%]
2024-07-09T10:14:46.4888780Z interpolate/tests/test_ndgriddata.py ....................                [ 15%]
2024-07-09T10:14:46.4956010Z interpolate/tests/test_pade.py ....                                      [ 15%]
2024-07-09T10:14:46.7352530Z interpolate/tests/test_polyint.py ...................................... [ 15%]
2024-07-09T10:14:46.8566020Z ........................................................................ [ 15%]
2024-07-09T10:14:46.8740440Z .................                                                        [ 15%]
2024-07-09T10:14:46.8957080Z interpolate/tests/test_rbf.py ..........                                 [ 15%]
2024-07-09T10:14:47.0738270Z interpolate/tests/test_rbfinterp.py .................................... [ 15%]
2024-07-09T10:14:47.2650430Z ........................................................................ [ 15%]
2024-07-09T10:14:47.5113650Z ........................................................................ [ 15%]
2024-07-09T10:14:47.5241690Z .....                                                                    [ 15%]
2024-07-09T10:14:48.0257510Z interpolate/tests/test_rgi.py ....................s..................... [ 15%]
2024-07-09T10:14:49.7347350Z ..................................ss.......ss..................s.......s [ 15%]
2024-07-09T10:14:50.5596830Z s.......ss............................................s.......ss........ [ 15%]
2024-07-09T10:14:50.6196080Z ..s.........................                                             [ 16%]
2024-07-09T10:14:50.6303250Z io/_harwell_boeing/tests/test_fortran_format.py ...........              [ 16%]
2024-07-09T10:14:50.6412170Z io/_harwell_boeing/tests/test_hb.py ..                                   [ 16%]
2024-07-09T10:14:50.8372350Z io/arff/tests/test_arffread.py ...........................               [ 16%]
2024-07-09T10:14:50.8392440Z io/matlab/tests/test_byteordercodes.py ..                                [ 16%]
2024-07-09T10:14:50.9405920Z io/matlab/tests/test_mio.py ............................................ [ 16%]
2024-07-09T10:14:50.9769900Z ............................                                             [ 16%]
2024-07-09T10:14:50.9829330Z io/matlab/tests/test_mio5_utils.py ......                                [ 16%]
2024-07-09T10:14:50.9840530Z io/matlab/tests/test_mio_funcs.py .                                      [ 16%]
2024-07-09T10:14:50.9857970Z io/matlab/tests/test_mio_utils.py ..                                     [ 16%]
2024-07-09T10:14:50.9872670Z io/matlab/tests/test_miobase.py .                                        [ 16%]
2024-07-09T10:14:50.9899830Z io/matlab/tests/test_pathological.py ..                                  [ 16%]
2024-07-09T10:14:51.6076130Z io/matlab/tests/test_streams.py ...........                              [ 16%]
2024-07-09T10:14:51.6357840Z io/tests/test_fortran.py ...........                                     [ 16%]
2024-07-09T10:14:51.6975860Z io/tests/test_idl.py ................................................... [ 16%]
2024-07-09T10:14:51.7507160Z ................                                                         [ 16%]
2024-07-09T10:14:51.8459390Z io/tests/test_mmio.py .................................................. [ 16%]
2024-07-09T10:14:52.0525170Z ........................................................................ [ 16%]
2024-07-09T10:14:52.1781320Z ..............................                                           [ 16%]
2024-07-09T10:14:52.2297610Z io/tests/test_netcdf.py .........................                        [ 16%]
2024-07-09T10:14:52.2471830Z io/tests/test_paths.py ...........                                       [ 16%]
2024-07-09T10:14:52.3103000Z io/tests/test_wavfile.py ................s.............................. [ 16%]
2024-07-09T10:14:52.4051870Z ........................................................................ [ 17%]
2024-07-09T10:14:52.5090840Z ........................................................................ [ 17%]
2024-07-09T10:14:52.6753370Z ........................................................................ [ 17%]
2024-07-09T10:14:52.7456120Z ..........................                                               [ 17%]
2024-07-09T10:14:52.7938530Z linalg/tests/test_basic.py ............................................. [ 17%]
2024-07-09T10:14:52.8787530Z ................................................................s....... [ 17%]
2024-07-09T10:14:52.9472940Z ........................................................................ [ 17%]
2024-07-09T10:14:53.4837720Z ........................................................................ [ 17%]
2024-07-09T10:14:54.3404830Z ........................................................................ [ 17%]
2024-07-09T10:14:54.4335140Z ........................................s............................... [ 18%]
2024-07-09T10:14:54.4576590Z ...........................                                              [ 18%]
2024-07-09T10:14:54.5152150Z linalg/tests/test_blas.py .............................................. [ 18%]
2024-07-09T10:14:54.5625230Z .....x.                                                                  [ 18%]
2024-07-09T10:14:54.5705570Z linalg/tests/test_cython_blas.py ......                                  [ 18%]
2024-07-09T10:14:54.5738780Z linalg/tests/test_cython_lapack.py ...                                   [ 18%]
2024-07-09T10:14:54.5977860Z linalg/tests/test_cythonized_array_utils.py ............................ [ 18%]
2024-07-09T10:14:54.6176280Z ......................                                                   [ 18%]
2024-07-09T10:14:54.7476860Z linalg/tests/test_decomp.py ............................................ [ 18%]
2024-07-09T10:14:54.8324210Z ........................................................................ [ 18%]
2024-07-09T10:14:54.9226810Z ........................................................................ [ 18%]
2024-07-09T10:14:58.3365110Z ........................................................................ [ 18%]
2024-07-09T10:14:58.4764580Z ........................................................................ [ 19%]
2024-07-09T10:14:58.5902640Z ........................................................................ [ 19%]
2024-07-09T10:14:59.3769020Z ..............................................x......................... [ 19%]
2024-07-09T10:14:59.4114420Z ................                                                         [ 19%]
2024-07-09T10:14:59.4436090Z linalg/tests/test_decomp_cholesky.py .....s............................. [ 19%]
2024-07-09T10:14:59.4787890Z ....................................                                     [ 19%]
2024-07-09T10:14:59.5673790Z linalg/tests/test_decomp_cossin.py ..................................... [ 19%]
2024-07-09T10:14:59.7591640Z ......................................................                   [ 19%]
2024-07-09T10:14:59.8045200Z linalg/tests/test_decomp_ldl.py ............                             [ 19%]
2024-07-09T10:14:59.8499530Z linalg/tests/test_decomp_lu.py ......................................... [ 19%]
2024-07-09T10:14:59.8973490Z ...................................................                      [ 19%]
2024-07-09T10:14:59.9048490Z linalg/tests/test_decomp_polar.py ..                                     [ 19%]
2024-07-09T10:14:59.9945930Z linalg/tests/test_decomp_update.py ..................................... [ 19%]
2024-07-09T10:15:00.1746640Z ........................................................................ [ 20%]
2024-07-09T10:15:00.3120800Z ........................................................................ [ 20%]
2024-07-09T10:15:00.4651120Z ........................................................................ [ 20%]
2024-07-09T10:15:00.6254490Z ........................................................................ [ 20%]
2024-07-09T10:15:00.8040730Z ........................................................................ [ 20%]
2024-07-09T10:15:00.9568510Z ........................................................................ [ 20%]
2024-07-09T10:15:01.1552150Z ........................................................................ [ 20%]
2024-07-09T10:15:01.3566930Z ........................................................................ [ 20%]
2024-07-09T10:15:01.4763820Z ....................................................                     [ 21%]
2024-07-09T10:15:06.5056140Z linalg/tests/test_extending.py .                                         [ 21%]
2024-07-09T10:15:06.5427880Z linalg/tests/test_fblas.py ............................................. [ 21%]
2024-07-09T10:15:06.6780290Z ........................................................................ [ 21%]
2024-07-09T10:15:06.6882240Z ............                                                             [ 21%]
2024-07-09T10:15:07.5339200Z linalg/tests/test_interpolative.py ..................................... [ 21%]
2024-07-09T10:15:07.5444960Z ..........                                                               [ 21%]
2024-07-09T10:15:07.5941610Z linalg/tests/test_lapack.py ............................................ [ 21%]
2024-07-09T10:15:07.6923310Z ....................................................................s... [ 21%]
2024-07-09T10:15:07.7727540Z ........................................................................ [ 21%]
2024-07-09T10:15:07.8527350Z ........................................................................ [ 21%]
2024-07-09T10:15:07.9207110Z ........................................................................ [ 22%]
2024-07-09T10:15:07.9951510Z ........................................................................ [ 22%]
2024-07-09T10:15:08.0691010Z ........................................................................ [ 22%]
2024-07-09T10:15:08.1576350Z ........................................................................ [ 22%]
2024-07-09T10:15:08.2409530Z ........................................................................ [ 22%]
2024-07-09T10:15:08.3137500Z ........................................................................ [ 22%]
2024-07-09T10:15:08.3907690Z ........................................................................ [ 22%]
2024-07-09T10:15:08.4672670Z ........................................................................ [ 22%]
2024-07-09T10:15:08.5464580Z ........................................................................ [ 23%]
2024-07-09T10:15:08.6297740Z ........................................................................ [ 23%]
2024-07-09T10:15:08.7047530Z ........................................................................ [ 23%]
2024-07-09T10:15:08.7771680Z ........................................................................ [ 23%]
2024-07-09T10:15:08.8509100Z ........................................................................ [ 23%]
2024-07-09T10:15:08.9244920Z ........................................................................ [ 23%]
2024-07-09T10:15:09.0135440Z ........................................................................ [ 23%]
2024-07-09T10:15:09.0994090Z ........................................................................ [ 24%]
2024-07-09T10:15:09.1688880Z ........................................................................ [ 24%]
2024-07-09T10:15:09.2433480Z ........................................................................ [ 24%]
2024-07-09T10:15:09.3176420Z ........................................................................ [ 24%]
2024-07-09T10:15:09.3871150Z ........................................................................ [ 24%]
2024-07-09T10:15:09.4503830Z ........................................................................ [ 24%]
2024-07-09T10:15:09.6009720Z ........................................................................ [ 24%]
2024-07-09T10:15:09.6778920Z ........................................................................ [ 25%]
2024-07-09T10:15:09.6901010Z ...........                                                              [ 25%]
2024-07-09T10:15:10.1008090Z linalg/tests/test_matfuncs.py ................xxxxx..............x...... [ 25%]
2024-07-09T10:15:10.7697250Z .............X........................                                   [ 25%]
2024-07-09T10:15:10.7775130Z linalg/tests/test_matmul_toeplitz.py ....                                [ 25%]
2024-07-09T10:15:10.7912770Z linalg/tests/test_procrustes.py ...........                              [ 25%]
2024-07-09T10:15:11.0652420Z linalg/tests/test_sketches.py .....                                      [ 25%]
2024-07-09T10:15:11.0950730Z linalg/tests/test_solve_toeplitz.py ......X.........................     [ 25%]
2024-07-09T10:15:11.3559020Z linalg/tests/test_solvers.py ........x....x....x.xx..................... [ 25%]
2024-07-09T10:15:11.3629790Z ....                                                                     [ 25%]
2024-07-09T10:15:11.4569850Z linalg/tests/test_special_matrices.py .................................. [ 25%]
2024-07-09T10:15:11.5297410Z ........................................................................ [ 25%]
2024-07-09T10:15:11.5357540Z ......                                                                   [ 25%]
2024-07-09T10:15:11.6376640Z misc/tests/test_common.py ...                                            [ 25%]
2024-07-09T10:15:11.6426770Z misc/tests/test_config.py ....                                           [ 25%]
2024-07-09T10:15:11.6471870Z misc/tests/test_doccer.py .....                                          [ 25%]
2024-07-09T10:15:11.6554370Z ndimage/tests/test_c_api.py ...                                          [ 25%]
2024-07-09T10:15:11.7703620Z ndimage/tests/test_datatypes.py .x                                       [ 25%]
2024-07-09T10:15:11.8099170Z ndimage/tests/test_filters.py .......................................... [ 25%]
2024-07-09T10:15:11.8746060Z ........................................................................ [ 25%]
2024-07-09T10:15:11.9408870Z ........................................................................ [ 25%]
2024-07-09T10:15:12.0060500Z ........................................................................ [ 26%]
2024-07-09T10:15:12.0687140Z ........................................................................ [ 26%]
2024-07-09T10:15:12.1318650Z ........................................................................ [ 26%]
2024-07-09T10:15:12.1954470Z ........................................................................ [ 26%]
2024-07-09T10:15:12.2615740Z ........................................................................ [ 26%]
2024-07-09T10:15:12.3250940Z ........................................................................ [ 26%]
2024-07-09T10:15:12.3928960Z ........................................................................ [ 26%]
2024-07-09T10:15:12.4606520Z ........................................................................ [ 27%]
2024-07-09T10:15:12.5452470Z ........................................................................ [ 27%]
2024-07-09T10:15:12.6220390Z ........................................................................ [ 27%]
2024-07-09T10:15:12.7023650Z ........................................................................ [ 27%]
2024-07-09T10:15:12.7842420Z ........................................................................ [ 27%]
2024-07-09T10:15:12.8742090Z ........................................................................ [ 27%]
2024-07-09T10:15:12.9463760Z ........................................................................ [ 27%]
2024-07-09T10:15:13.0147990Z ........................................................................ [ 27%]
2024-07-09T10:15:13.0845290Z ........................................................................ [ 28%]
2024-07-09T10:15:13.1497550Z ........................................................................ [ 28%]
2024-07-09T10:15:13.2107280Z ........................................................................ [ 28%]
2024-07-09T10:15:13.2720800Z ........................................................................ [ 28%]
2024-07-09T10:15:13.3373750Z ........................................................................ [ 28%]
2024-07-09T10:15:13.4042240Z ........................................................................ [ 28%]
2024-07-09T10:15:13.4707980Z ........................................................................ [ 28%]
2024-07-09T10:15:13.5378050Z ........................................................................ [ 29%]
2024-07-09T10:15:13.6017690Z ........................................................................ [ 29%]
2024-07-09T10:15:14.0162760Z .........................................................                [ 29%]
2024-07-09T10:15:14.0562490Z ndimage/tests/test_fourier.py .......................................... [ 29%]
2024-07-09T10:15:14.0848060Z ..................................                                       [ 29%]
2024-07-09T10:15:14.1189730Z ndimage/tests/test_interpolation.py .................................... [ 29%]
2024-07-09T10:15:14.1818590Z ........................................................................ [ 29%]
2024-07-09T10:15:14.2420070Z ........................................................................ [ 29%]
2024-07-09T10:15:14.3013030Z ........................................................................ [ 29%]
2024-07-09T10:15:14.3611000Z ........................................................................ [ 30%]
2024-07-09T10:15:14.4311920Z ........................................................................ [ 30%]
2024-07-09T10:15:14.5121480Z ........................................................................ [ 30%]
2024-07-09T10:15:14.5722300Z ........................................................................ [ 30%]
2024-07-09T10:15:14.6336240Z ........................................................................ [ 30%]
2024-07-09T10:15:14.6953700Z ........................................................................ [ 30%]
2024-07-09T10:15:14.7701420Z ........................................................................ [ 30%]
2024-07-09T10:15:14.8486430Z ........................................................................ [ 31%]
2024-07-09T10:15:14.9185810Z ........................................................................ [ 31%]
2024-07-09T10:15:14.9838040Z ........................................................................ [ 31%]
2024-07-09T10:15:14.9980120Z ................                                                         [ 31%]
2024-07-09T10:15:15.1718370Z ndimage/tests/test_measurements.py ..................................... [ 31%]
2024-07-09T10:15:15.2533650Z ........................................................................ [ 31%]
2024-07-09T10:15:15.2672510Z ................                                                         [ 31%]
2024-07-09T10:15:15.3084360Z ndimage/tests/test_morphology.py ....................................... [ 31%]
2024-07-09T10:15:15.3795670Z ........................................................................ [ 31%]
2024-07-09T10:15:15.4443610Z ........................................................................ [ 31%]
2024-07-09T10:15:15.5042860Z ........................................................................ [ 32%]
2024-07-09T10:15:15.5675510Z ........................................................................ [ 32%]
2024-07-09T10:15:15.6323520Z ........................................................................ [ 32%]
2024-07-09T10:15:15.6942970Z ........................................................................ [ 32%]
2024-07-09T10:15:15.7532250Z ........................................................................ [ 32%]
2024-07-09T10:15:15.8135430Z ........................................................................ [ 32%]
2024-07-09T10:15:15.8747790Z ........................................................................ [ 32%]
2024-07-09T10:15:15.9362540Z ........................................................................ [ 32%]
2024-07-09T10:15:15.9974380Z ........................................................................ [ 33%]
2024-07-09T10:15:16.0146030Z .................                                                        [ 33%]
2024-07-09T10:15:16.0230610Z ndimage/tests/test_ni_support.py ..........                              [ 33%]
2024-07-09T10:15:16.0489620Z ndimage/tests/test_splines.py ..................                         [ 33%]
2024-07-09T10:15:16.0801220Z odr/tests/test_odr.py ................s..                                [ 33%]
2024-07-09T10:15:16.0919140Z optimize/_trustregion_constr/tests/test_canonical_constraint.py ......   [ 33%]
2024-07-09T10:15:16.1229470Z optimize/_trustregion_constr/tests/test_projections.py ..........        [ 33%]
2024-07-09T10:15:16.1436180Z optimize/_trustregion_constr/tests/test_qp_subproblem.py ............... [ 33%]
2024-07-09T10:15:16.1563350Z ........                                                                 [ 33%]
2024-07-09T10:15:16.1774570Z optimize/_trustregion_constr/tests/test_report.py .s                     [ 33%]
2024-07-09T10:15:18.1183910Z optimize/tests/test__basinhopping.py .................................   [ 33%]
2024-07-09T10:15:18.4623840Z optimize/tests/test__differential_evolution.py ......................... [ 33%]
2024-07-09T10:15:22.9791320Z .............................s.......                                    [ 33%]
2024-07-09T10:15:25.3800790Z optimize/tests/test__dual_annealing.py ................................. [ 33%]
2024-07-09T10:15:26.8269010Z .............                                                            [ 33%]
2024-07-09T10:15:26.8392760Z optimize/tests/test__linprog_clean_inputs.py .............               [ 33%]
2024-07-09T10:15:26.9077750Z optimize/tests/test__numdiff.py ................................         [ 33%]
2024-07-09T10:15:26.9533350Z optimize/tests/test__remove_redundancy.py .............................. [ 33%]
2024-07-09T10:15:27.0364490Z ......................................                                   [ 33%]
2024-07-09T10:15:27.0545480Z optimize/tests/test__root.py ...............                             [ 33%]
2024-07-09T10:15:28.4213460Z optimize/tests/test__shgo.py ..................ss...................s... [ 33%]
2024-07-09T10:15:29.7282670Z ............................                                             [ 33%]
2024-07-09T10:15:29.8347330Z optimize/tests/test__spectral.py ....                                    [ 33%]
2024-07-09T10:15:29.8976010Z optimize/tests/test_bracket.py ......................................... [ 34%]
2024-07-09T10:15:30.0832880Z ........................................................................ [ 34%]
2024-07-09T10:15:30.1558930Z ........................................................................ [ 34%]
2024-07-09T10:15:30.2331280Z ........................................................................ [ 34%]
2024-07-09T10:15:30.3187730Z ........................................................................ [ 34%]
2024-07-09T10:15:30.6099050Z ........................................................................ [ 34%]
2024-07-09T10:15:30.7122890Z ...................................................................      [ 34%]
2024-07-09T10:15:31.1437480Z optimize/tests/test_chandrupatla.py .................................... [ 34%]
2024-07-09T10:15:31.3160680Z ........................................................................ [ 35%]
2024-07-09T10:15:31.3628760Z ...........................                                              [ 35%]
2024-07-09T10:15:31.3758290Z optimize/tests/test_cobyla.py s.....                                     [ 35%]
2024-07-09T10:15:31.7728800Z optimize/tests/test_cobyqa.py ........                                   [ 35%]
2024-07-09T10:15:32.7723680Z optimize/tests/test_constraint_conversion.py ........                    [ 35%]
2024-07-09T10:15:32.7888780Z optimize/tests/test_constraints.py ................                      [ 35%]
2024-07-09T10:15:32.7930720Z optimize/tests/test_cython_optimize.py .....                             [ 35%]
2024-07-09T10:15:32.8319130Z optimize/tests/test_differentiable_functions.py ..............x.         [ 35%]
2024-07-09T10:15:32.9012880Z optimize/tests/test_differentiate.py ................................... [ 35%]
2024-07-09T10:15:33.2916880Z ........................................................................ [ 35%]
2024-07-09T10:15:33.4273250Z ...............................................XX...................     [ 35%]
2024-07-09T10:15:33.5483090Z optimize/tests/test_direct.py ....................................ss.... [ 35%]
2024-07-09T10:15:33.5655220Z .....................                                                    [ 35%]
2024-07-09T10:15:36.2213890Z optimize/tests/test_extending.py .                                       [ 35%]
2024-07-09T10:15:36.2360910Z optimize/tests/test_hessian_update_strategy.py .....                     [ 35%]
2024-07-09T10:15:36.2642970Z optimize/tests/test_isotonic_regression.py ............................. [ 35%]
2024-07-09T10:15:36.2786870Z ................                                                         [ 35%]
2024-07-09T10:15:36.3305210Z optimize/tests/test_lbfgsb_hessinv.py ..                                 [ 35%]
2024-07-09T10:15:36.3327930Z optimize/tests/test_lbfgsb_setulb.py ..                                  [ 35%]
2024-07-09T10:15:36.9080380Z optimize/tests/test_least_squares.py ................................... [ 35%]
2024-07-09T10:15:38.1155430Z ........................................................................ [ 35%]
2024-07-09T10:15:38.1291890Z ...........                                                              [ 35%]
2024-07-09T10:15:38.1900150Z optimize/tests/test_linear_assignment.py ......................          [ 36%]
2024-07-09T10:15:38.2060650Z optimize/tests/test_linesearch.py ...........                            [ 36%]
2024-07-09T10:15:38.4155820Z optimize/tests/test_linprog.py ......................................... [ 36%]
2024-07-09T10:15:38.6611000Z .......................................s.ss............................. [ 36%]
2024-07-09T10:15:39.0405520Z ...............................................s.s...................... [ 36%]
2024-07-09T10:15:39.3690070Z ....................................................s...s.ss............ [ 36%]
2024-07-09T10:15:39.4870250Z ................................................................ss...... [ 36%]
2024-07-09T10:15:39.8017070Z .....................................................................ssX [ 36%]
2024-07-09T10:15:40.2077170Z ........................................................................ [ 36%]
2024-07-09T10:15:40.3654850Z ....sssX................................................................ [ 37%]
2024-07-09T10:15:40.4783920Z .............s.ssss..................................................... [ 37%]
2024-07-09T10:15:40.6751150Z ..........................s.sss..............................x.......... [ 37%]
2024-07-09T10:15:40.9290110Z ..................................s..................................... [ 37%]
2024-07-09T10:15:42.1105030Z ........x............................................s.................. [ 37%]
2024-07-09T10:15:42.1674890Z .s...........................                                            [ 37%]
2024-07-09T10:15:42.1775110Z optimize/tests/test_lsq_common.py ..........                             [ 37%]
2024-07-09T10:15:43.1575130Z optimize/tests/test_lsq_linear.py ..........s........s...                [ 37%]
2024-07-09T10:15:43.2816280Z optimize/tests/test_milp.py .x.........s                                 [ 37%]
2024-07-09T10:15:43.5628820Z optimize/tests/test_minimize_constrained.py .....................s...... [ 37%]
2024-07-09T10:15:44.1901520Z ..........s.............................ssssssssssssssssssssssssssssssss [ 37%]
2024-07-09T10:15:45.1899620Z ss.....................s................s............................... [ 38%]
2024-07-09T10:15:46.3663690Z ..s.......x........s.................................s.......x........s. [ 38%]
2024-07-09T10:15:46.7592070Z .....................x........X                                          [ 38%]
2024-07-09T10:15:46.8312760Z optimize/tests/test_minpack.py ......................................... [ 38%]
2024-07-09T10:15:46.9424900Z ....................................................                     [ 38%]
2024-07-09T10:15:46.9556820Z optimize/tests/test_nnls.py .......                                      [ 38%]
2024-07-09T10:15:47.1980860Z optimize/tests/test_nonlin.py .......................................    [ 38%]
2024-07-09T10:15:47.6160690Z optimize/tests/test_optimize.py .........................x.............. [ 38%]
2024-07-09T10:15:48.1526430Z ...........x.........................x.........................x........ [ 38%]
2024-07-09T10:15:49.6752560Z .....................................................x.................. [ 38%]
2024-07-09T10:15:50.1991420Z ........................................................................ [ 39%]
2024-07-09T10:15:50.3351140Z ........................................................................ [ 39%]
2024-07-09T10:15:58.4262230Z ......................................................s.ss.ss.ss.ss.ss.s [ 39%]
2024-07-09T10:15:58.5769120Z s.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.s [ 39%]
2024-07-09T10:15:58.7756750Z s.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.s.................................... [ 39%]
2024-07-09T10:15:58.9463060Z s.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.s [ 39%]
2024-07-09T10:15:59.2881390Z s.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.ss.s.................. [ 39%]
2024-07-09T10:15:59.3500620Z ....                                                                     [ 39%]
2024-07-09T10:15:59.3950930Z optimize/tests/test_quadratic_assignment.py ..............               [ 39%]
2024-07-09T10:15:59.3976070Z optimize/tests/test_regression.py ...                                    [ 39%]
2024-07-09T10:15:59.4530400Z optimize/tests/test_slsqp.py .....................................       [ 39%]
2024-07-09T10:15:59.5016050Z optimize/tests/test_tnc.py ....................                          [ 39%]
2024-07-09T10:15:59.5339520Z optimize/tests/test_trustregion.py ........                              [ 39%]
2024-07-09T10:15:59.8097390Z optimize/tests/test_trustregion_exact.py ..........                      [ 40%]
2024-07-09T10:15:59.8149600Z optimize/tests/test_trustregion_krylov.py .....                          [ 40%]
2024-07-09T10:15:59.8613950Z optimize/tests/test_zeros.py ........................................... [ 40%]
2024-07-09T10:16:00.0879160Z ........................................................................ [ 40%]
2024-07-09T10:16:00.1109110Z ......................                                                   [ 40%]
2024-07-09T10:16:00.1167350Z signal/tests/test_array_tools.py ......                                  [ 40%]
2024-07-09T10:16:00.1469750Z signal/tests/test_bsplines.py ......................XXXXXXX..            [ 40%]
2024-07-09T10:16:00.1739540Z signal/tests/test_cont2discrete.py ........................              [ 40%]
2024-07-09T10:16:00.2598200Z signal/tests/test_czt.py ............................................... [ 40%]
2024-07-09T10:16:00.3353940Z ........................................................................ [ 40%]
2024-07-09T10:16:00.3776900Z ..........................................                               [ 40%]
2024-07-09T10:16:00.4092980Z signal/tests/test_dltisys.py ................................            [ 40%]
2024-07-09T10:16:00.4527410Z signal/tests/test_filter_design.py ..................................... [ 40%]
2024-07-09T10:16:00.7298940Z ........................................................................ [ 40%]
2024-07-09T10:16:01.8037600Z ........................................................................ [ 41%]
2024-07-09T10:16:02.0839580Z ..............................................................           [ 41%]
2024-07-09T10:16:02.1196630Z signal/tests/test_fir_filter_design.py ................................. [ 41%]
2024-07-09T10:16:02.2388470Z ....                                                                     [ 41%]
2024-07-09T10:16:02.3864560Z signal/tests/test_ltisys.py ............................................ [ 41%]
2024-07-09T10:16:02.4179210Z .....................................                                    [ 41%]
2024-07-09T10:16:02.4248210Z signal/tests/test_max_len_seq.py ..                                      [ 41%]
2024-07-09T10:16:02.4607540Z signal/tests/test_peak_finding.py ...................................... [ 41%]
2024-07-09T10:16:02.5206730Z ...............                                                          [ 41%]
2024-07-09T10:16:02.5257490Z signal/tests/test_result_type.py .....                                   [ 41%]
2024-07-09T10:16:02.5515660Z signal/tests/test_savitzky_golay.py ..............                       [ 41%]
2024-07-09T10:16:02.6127120Z signal/tests/test_short_time_fft.py .................................... [ 41%]
2024-07-09T10:16:02.8062180Z ........................................................................ [ 41%]
2024-07-09T10:16:03.1699940Z ........................................                                 [ 41%]
2024-07-09T10:16:03.2585410Z signal/tests/test_signaltools.py ....................................... [ 41%]
2024-07-09T10:16:03.3188100Z ........................................................................ [ 42%]
2024-07-09T10:16:03.5789530Z ........................................................................ [ 42%]
2024-07-09T10:16:03.6483680Z ........................................................................ [ 42%]
2024-07-09T10:16:03.7216200Z ........................................................................ [ 42%]
2024-07-09T10:16:03.7953200Z ........................................................................ [ 42%]
2024-07-09T10:16:03.8685520Z ........................................................................ [ 42%]
2024-07-09T10:16:03.9417370Z ........................................................................ [ 42%]
2024-07-09T10:16:04.0166170Z ........................................................................ [ 43%]
2024-07-09T10:16:04.1058930Z ........................................................................ [ 43%]
2024-07-09T10:16:04.2119420Z ........................................................................ [ 43%]
2024-07-09T10:16:04.3251090Z ........................................................................ [ 43%]
2024-07-09T10:16:04.4468720Z ........................................................................ [ 43%]
2024-07-09T10:16:04.5551980Z ........................................................................ [ 43%]
2024-07-09T10:16:04.6493940Z ........................................................................ [ 43%]
2024-07-09T10:16:04.7529190Z ........................................................................ [ 43%]
2024-07-09T10:16:04.8595980Z ........................................................................ [ 44%]
2024-07-09T10:16:04.9487580Z ........................................................................ [ 44%]
2024-07-09T10:16:04.9614450Z .....ss..../Users/runner/work/scipy/scipy/tools/wheels/cibw_test_command.sh: line 11: 13615 Trace/BPT trap: 5       python -c "import sys; import scipy; sys.exit(not scipy.test())"
2024-07-09T10:16:05.8676330Z ##[endgroup]
2024-07-09T10:16:05.8678070Z                                                             [31m✕ [0m121.80s
2024-07-09T10:16:05.8724980Z ##[error]Command ['/bin/sh', '-c', 'bash /Users/runner/work/scipy/scipy/tools/wheels/cibw_test_command.sh /Users/runner/work/scipy/scipy'] failed with code 133. 
2024-07-09T10:16:05.8731550Z 
2024-07-09T10:16:05.9060960Z ##[error]Process completed with exit code 1.
2024-07-09T10:16:05.9228700Z Post job cleanup.
2024-07-09T10:16:05.9299200Z Post job cleanup.
2024-07-09T10:16:06.2955200Z [command]/opt/homebrew/bin/git version
2024-07-09T10:16:06.3255860Z git version 2.45.2
2024-07-09T10:16:06.3296050Z Copying '/Users/runner/.gitconfig' to '/Users/runner/work/_temp/955ef7cb-151e-4e43-9d1d-139ce053d906/.gitconfig'
2024-07-09T10:16:06.3306900Z Temporarily overriding HOME='/Users/runner/work/_temp/955ef7cb-151e-4e43-9d1d-139ce053d906' before making global git config changes
2024-07-09T10:16:06.3307910Z Adding repository directory to the temporary git global config as a safe directory
2024-07-09T10:16:06.3309400Z [command]/opt/homebrew/bin/git config --global --add safe.directory /Users/runner/work/scipy/scipy
2024-07-09T10:16:06.3536820Z [command]/opt/homebrew/bin/git config --local --name-only --get-regexp core\.sshCommand
2024-07-09T10:16:06.4118110Z [command]/opt/homebrew/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2024-07-09T10:16:06.5565800Z Entering 'doc/source/_static/scipy-mathjax'
2024-07-09T10:16:06.5729790Z Entering 'scipy/_lib/array_api_compat'
2024-07-09T10:16:06.5886670Z Entering 'scipy/_lib/boost_math'
2024-07-09T10:16:06.6096750Z Entering 'scipy/_lib/cobyqa'
2024-07-09T10:16:06.6270050Z Entering 'scipy/_lib/highs'
2024-07-09T10:16:06.6455590Z Entering 'scipy/_lib/pocketfft'
2024-07-09T10:16:06.6631580Z Entering 'scipy/_lib/unuran'
2024-07-09T10:16:06.6809800Z Entering 'scipy/sparse/linalg/_propack/PROPACK'
2024-07-09T10:16:06.6968700Z [command]/opt/homebrew/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2024-07-09T10:16:06.7004900Z http.https://github.com/.extraheader
2024-07-09T10:16:06.7012330Z [command]/opt/homebrew/bin/git config --local --unset-all http.https://github.com/.extraheader
2024-07-09T10:16:06.7057840Z [command]/opt/homebrew/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2024-07-09T10:16:06.7621440Z Entering 'doc/source/_static/scipy-mathjax'
2024-07-09T10:16:06.7718540Z http.https://github.com/.extraheader
2024-07-09T10:16:06.7813910Z Entering 'scipy/_lib/array_api_compat'
2024-07-09T10:16:06.7932730Z http.https://github.com/.extraheader
2024-07-09T10:16:06.8060740Z Entering 'scipy/_lib/boost_math'
2024-07-09T10:16:06.8171270Z http.https://github.com/.extraheader
2024-07-09T10:16:06.8279550Z Entering 'scipy/_lib/cobyqa'
2024-07-09T10:16:06.8394170Z http.https://github.com/.extraheader
2024-07-09T10:16:06.8489320Z Entering 'scipy/_lib/highs'
2024-07-09T10:16:06.8592700Z http.https://github.com/.extraheader
2024-07-09T10:16:06.8706510Z Entering 'scipy/_lib/pocketfft'
2024-07-09T10:16:06.8956800Z http.https://github.com/.extraheader
2024-07-09T10:16:06.9190440Z Entering 'scipy/_lib/unuran'
2024-07-09T10:16:06.9285650Z http.https://github.com/.extraheader
2024-07-09T10:16:06.9469110Z Entering 'scipy/sparse/linalg/_propack/PROPACK'
2024-07-09T10:16:06.9562200Z http.https://github.com/.extraheader
2024-07-09T10:16:06.9945710Z Cleaning up orphan processes
2024-07-09T10:16:07.7179740Z Terminate orphan process: pid (4062) (diskimages-help)

```

</details>

Based on the test output (`......ss....`), and assuming that locally the test order within `test_signaltools.py` is unchanged from what it was in CI, the crash happened in:
```
scipy/signal/tests/test_signaltools.py::TestMedFilt::test_medfilt2d_parallel[float32]
```
which certainly looks likely since it's calling `medfilt2d` () under `ThreadPoolExecutor(max_workers=4)`: 
https://github.com/scipy/scipy/blob/81c53d48a290b604ec5faa34c0a7d48537b487d6/scipy/signal/tests/test_signaltools.py#L1171-L1180

The C code that ends up being called there is:
- `_sigtools_median2d`: https://github.com/scipy/scipy/blob/81c53d48a290b604ec5faa34c0a7d48537b487d6/scipy/signal/_sigtoolsmodule.c#L954
- `_f_medfilt2d`: https://github.com/scipy/scipy/blob/81c53d48a290b604ec5faa34c0a7d48537b487d6/scipy/signal/_medianfilter.c#L130

I'd expect this to be reproducible locally.

--- a/scipy/signal/_medianfilter.c
+++ b/scipy/signal/_medianfilter.c
@@ -7,10 +7,9 @@
 
 
 /* defined below */
-void f_medfilt2(float*,float*,npy_intp*,npy_intp*);
-void d_medfilt2(double*,double*,npy_intp*,npy_intp*);
-void b_medfilt2(unsigned char*,unsigned char*,npy_intp*,npy_intp*);
-extern char *check_malloc (size_t);
+void f_medfilt2(float*,float*,npy_intp*,npy_intp*,int*);
+void d_medfilt2(double*,double*,npy_intp*,npy_intp*,int*);
+void b_medfilt2(unsigned char*,unsigned char*,npy_intp*,npy_intp*,int*);
 
 
 /* The QUICK_SELECT routine is based on Hoare's Quickselect algorithm,
@@ -72,7 +71,7 @@ TYPE NAME(TYPE arr[], int n)                                            \
 
 /* 2-D median filter with zero-padding on edges. */
 #define MEDIAN_FILTER_2D(NAME, TYPE, SELECT)                            \
-void NAME(TYPE* in, TYPE* out, npy_intp* Nwin, npy_intp* Ns)                    \
+void NAME(TYPE* in, TYPE* out, npy_intp* Nwin, npy_intp* Ns, int* errnum)    \
 {                                                                       \
     int nx, ny, hN[2];                                                  \
     int pre_x, pre_y, pos_x, pos_y;                                     \
@@ -80,7 +79,11 @@ void NAME(TYPE* in, TYPE* out, npy_intp* Nwin, npy_intp* Ns)                    
     TYPE *myvals, *fptr1, *fptr2, *ptr1, *ptr2;                         \
                                                                         \
     totN = Nwin[0] * Nwin[1];                                           \
-    myvals = (TYPE *) check_malloc( totN * sizeof(TYPE));               \
+    myvals = (TYPE *) malloc( totN * sizeof(TYPE));                     \
+    if (myvals == NULL) {                                               \
+	*errnum = -1;                                                   \
+	return;                                                         \
+    }                                                                   \
                                                                         \
     Py_BEGIN_ALLOW_THREADS                                              \
                                                                         \
@@ -118,6 +121,7 @@ void NAME(TYPE* in, TYPE* out, npy_intp* Nwin, npy_intp* Ns)                    
     Py_END_ALLOW_THREADS                                                \
                                                                         \
     free(myvals);                                                       \
+    *errnum = 0;                                                        \
 }
 
 

--- a/scipy/signal/_sigtoolsmodule.c
+++ b/scipy/signal/_sigtoolsmodule.c
@@ -10,23 +10,9 @@ is granted under the SciPy License.
 #include "npy_2_compat.h"
 
 #include "_sigtools.h"
-#include <setjmp.h>
 #include <stdlib.h>
 
 #define PYERR(message) {PyErr_SetString(PyExc_ValueError, message); goto fail;}
-
-
-jmp_buf MALLOC_FAIL;
-
-char *check_malloc(size_t size)
-{
-    char *the_block = malloc(size);
-    if (the_block == NULL) {
-        printf("\nERROR: unable to allocate %zu bytes!\n", size);
-        longjmp(MALLOC_FAIL,-1);
-    }
-    return the_block;
-}
 
 
 /********************************************************
@@ -947,14 +933,14 @@ fail:
 
 static char doc_median2d[] = "filt = _median2d(data, size)";
 
-extern void f_medfilt2(float*,float*,npy_intp*,npy_intp*);
-extern void d_medfilt2(double*,double*,npy_intp*,npy_intp*);
-extern void b_medfilt2(unsigned char*,unsigned char*,npy_intp*,npy_intp*);
+extern void f_medfilt2(float*,float*,npy_intp*,npy_intp*,int*);
+extern void d_medfilt2(double*,double*,npy_intp*,npy_intp*,int*);
+extern void b_medfilt2(unsigned char*,unsigned char*,npy_intp*,npy_intp*,int*);
 
 static PyObject *_sigtools_median2d(PyObject *NPY_UNUSED(dummy), PyObject *args)
 {
     PyObject *image=NULL, *size=NULL;
-    int typenum;
+    int typenum, errnum=-2;
     PyArrayObject *a_image=NULL, *a_size=NULL;
     PyArrayObject *a_out=NULL;
     npy_intp Nwin[2] = {3,3};
@@ -977,29 +963,30 @@ static PyObject *_sigtools_median2d(PyObject *NPY_UNUSED(dummy), PyObject *args)
     a_out = (PyArrayObject *)PyArray_SimpleNew(2, PyArray_DIMS(a_image), typenum);
     if (a_out == NULL) goto fail;
 
-    if (setjmp(MALLOC_FAIL)) {
-	PYERR("Memory allocation error.");
-    }
-    else {
-	switch (typenum) {
-	case NPY_UBYTE:
-	    b_medfilt2((unsigned char *)PyArray_DATA(a_image),
+    switch (typenum) {
+        case NPY_UBYTE:
+            b_medfilt2((unsigned char *)PyArray_DATA(a_image),
                        (unsigned char *)PyArray_DATA(a_out),
-                       Nwin, PyArray_DIMS(a_image));
-	    break;
-	case NPY_FLOAT:
-	    f_medfilt2((float *)PyArray_DATA(a_image),
+                       Nwin, PyArray_DIMS(a_image),
+                       &errnum);
+            break;
+        case NPY_FLOAT:
+            f_medfilt2((float *)PyArray_DATA(a_image),
                        (float *)PyArray_DATA(a_out), Nwin,
-                       PyArray_DIMS(a_image));
-	    break;
-	case NPY_DOUBLE:
-	    d_medfilt2((double *)PyArray_DATA(a_image),
+                       PyArray_DIMS(a_image),
+                       &errnum);
+            break;
+        case NPY_DOUBLE:
+            d_medfilt2((double *)PyArray_DATA(a_image),
                        (double *)PyArray_DATA(a_out), Nwin,
-                       PyArray_DIMS(a_image));
-	    break;
-	default:
-	  PYERR("2D median filter only supports uint8, float32, and float64.");
-	}
+                       PyArray_DIMS(a_image),
+                       &errnum);
+            break;
+        default:
+            PYERR("2D median filter only supports uint8, float32, and float64.");
+    }
+    if (errnum != 0) {
+        PYERR("ERROR: unable to allocate enough memory in _medfilt2d!\n");
     }
 
     Py_DECREF(a_image);

